apps/system/js/devtools/devtools_auth.js

```diff
@@ -0,0 +1,176 @@
+   * place when debugging via WiFi to exchange keys.
+   * @requires DevToolsAuthDialog
+   * @requires ScreenManager
+   * @class DevToolsAuth
+   */
+  function DevToolsAuth() {
+    window.addEventListener('mozChromeEvent', this);
+  }
+
+  DevToolsAuth.prototype = {
+
+    /**
+     * General event handler interface.
+     * Displays the pair dialog when needed.
+     * @memberof DevToolsAuth.prototype
+     * @param  {DOMEvent} evt The event.
+     */
+    handleEvent: function(e) {
+      if (e.detail.type !== 'devtools-auth') {
+        return;
+      }
+      if (e.detail.action === 'start') {
+        this.start();
+      }
+      if (e.detail.action === 'capture') {
+        this.capture();
+      }
+      if (e.detail.action === 'stop') {
+        this.stop();
+      }
+    },
+
+    get dialog() {
+      if (!this._dialog) {
+        this._dialog = new DevToolsAuthDialog({
+          onHide: this.onHide.bind(this)
+        });
+      }
+      return this._dialog;
+    },
+
+    get canvas() {
+      if (!this._canvas) {
+        this._canvas = document.createElement('canvas');
+      }
+      return this._canvas;
+    },
+
+    /**
+     * Begin capturing images that DevTools will attempt to find a QR code in
+     * for authentication.
+     * @memberof DevToolsAuth.prototype
+     */
+    start: function() {
+      // We want the user's attention, so we need to turn the screen on if it's
+      // off.
+      if (!ScreenManager.screenEnabled) {
+        ScreenManager.turnScreenOn();
+      }
+
+      this.dialog.show();
+
+      var getUserMedia = navigator.getUserMedia || navigator.mozGetUserMedia;
+      getUserMedia = getUserMedia.bind(navigator);
+
+      getUserMedia({ video: true, audio: false }, function(stream) {
+        var video = this.dialog.video;
+
+        video.addEventListener('loadedmetadata', function onMetadata() {
+          video.removeEventListener('loadedmetadata', onMetadata);
+          this._resizeCanvas();
+          this.capture();
+        }.bind(this));
+
+        video.mozSrcObject = stream;
+        video.play();
+      }.bind(this), function(e) {
+        console.error('Unable to access camera: ' + e);
+      });
+    },
+
+    /**
+     * Capture another image and send to DevTools.
+     * @memberof DevToolsAuth.prototype
+     */
+    capture: function() {
+      if (this._inflightURL) {
+        URL.revokeObjectURL(this._inflightURL);
+        this._inflightURL = null;
+      }
+      this._captureTimeout = setTimeout(function() {
+        var video = this.dialog.video;
+        var canvas = this.canvas;
+        var context = canvas.getContext('2d');
+        context.drawImage(video, 0, 0, canvas.width, canvas.height);
+        canvas.toBlob(function(blob) {
+          this._inflightURL = URL.createObjectURL(blob);
+          this._checkImage(this._inflightURL);
+        }.bind(this), 'image/jpeg');
+      }.bind(this), 2000);
```

Comment:
I've removed this timeout. Capture goes faster without, and the phone doesn't seem to explode or anything. :wink: